core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestRewriteCaseExpressionPredicate.java

```diff
@@ -0,0 +1,266 @@
+                "('case1' = 'case1' AND col1 = 1) OR ('case2' = 'case1' AND col1 = 2) OR ('default' = 'case1' AND (NOT(col1 = 1) AND NOT(col1 = 2)))",
+                "col1 = 1");
+
+        assertRewrittenAndOptimizedExpression(
+                "(case when col3='a' then 'case1' when col3='b' then 'case2' else 'default' end) = 'case1'",
+                "('case1' = 'case1' AND col3 = 'a') OR ('case2' = 'case1' AND col3 = 'b') OR ('default' = 'case1' AND (NOT(col3 = 'a') AND NOT(col3 = 'b')))",
+                "col3 = 'a'");
+
+        assertRewrittenAndOptimizedExpression(
+                "(case when col1=1 then 'case1' when col1=2 then 'case2' else 'default' end) = 'default'",
+                "('case1' = 'default' AND col1 = 1) OR ('case2' = 'default' AND col1 = 2) OR ('default' = 'default' AND (NOT(col1 = 1) AND NOT(col1 = 2)))",
+                "col1 NOT IN (1, 2)");
+    }
+
+    @Test
+    public void testRewriterOnCaseExpressionInRightSideOfComparisonFunction()
+    {
+        assertRewrittenAndOptimizedExpression(
+                "(case col1 when 1 then 10 when 2 then 20 else 30 end) > 20",
+                "(10 > 20 AND col1 = 1) OR (20 > 20 AND col1 = 2) OR (30 > 20 AND (NOT(col1 = 1) AND NOT(col1 = 2)))",
+                "col1 NOT IN (1, 2)");
+
+        assertRewrittenAndOptimizedExpression(
+                "25 < (case col1 when 1 then 10 when 2 then 20 else 30 end)",
+                "(25 < 10 AND col1 = 1) OR (25 < 20 AND col1 = 2) OR (25 < 30 AND (NOT(col1 = 1) AND NOT(col1 = 2)))",
+                "col1 NOT IN (1, 2)");
+    }
+
+    @Test
+    public void testRewriterWhenMoreThanOneConditionMatches()
+    {
+        assertRewrittenAndOptimizedExpression(
+                "(case col1 when 1 then 'case' when 2 then 'case' else 'default' end) = 'case'",
+                "('case' = 'case' AND col1 = 1) OR ('case' = 'case' AND col1 = 2) OR ('default' = 'case' AND (NOT(col1 = 1) AND NOT(col1 = 2)))",
+                "col1 IN (1, 2)");
+
+        assertRewrittenAndOptimizedExpression(
+                "(case col1 when 1 then 'defaultAndCase1' when 2 then 'case2' else 'defaultAndCase1' end) = 'defaultAndCase1'",
+                "('defaultAndCase1' = 'defaultAndCase1' AND col1 = 1) OR ('case2' = 'defaultAndCase1' AND col1 = 2) OR ('defaultAndCase1' = 'defaultAndCase1' AND (NOT(col1 = 1) AND NOT(col1 = 2)))",
+                "col1 <> 2");
+
+        assertRewrittenAndOptimizedExpression(
+                "(case col3 when 'data1' then 'case1' when 'data2' then 'case2' else col3 end) = 'case1'",
+                "('case1' = 'case1' AND col3 = 'data1') OR ('case2' = 'case1' AND col3 = 'data2') OR (col3 = 'case1' AND (NOT(col3 = 'data1') AND NOT(col3 = 'data2')))",
+                "col3 IN (VARCHAR 'case1', VARCHAR 'data1')");
+    }
+
+    @Test(enabled = false)
+    public void testRewriterOnCaseExpressionWithoutElseClause()
```

Comment:
We need to get it tested somehow. Have you tried to follow my suggestion https://github.com/trinodb/trino/pull/10580#discussion_r791604255?

Reply:
Done.